source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/teamexplorer/internal/TeamExplorerHelpers.java

```diff
@@ -279,4 +311,219 @@ public static IViewPart showView(final String viewID) {
 
         return view;
     }
+
+    public static boolean tryParseProtocolHandlerArguments(final String[] applicationArgs) {
+
+        if (applicationArgs == null) {
+            return false;
+        }
+
+        boolean found = false;
+
+        for (final String arg : applicationArgs) {
+            if (found) {
+                final URI uri = URIUtils.newURI(arg);
+
+                if (!uri.getScheme().equalsIgnoreCase(PROTOCOL_HANDLER_SCHEME)) {
+                    found = false;
+                    continue;
+                }
+
+                boolean repoUrlAvailable = false;
+                final String[] queryItems = uri.getQuery().split("&"); //$NON-NLS-1$
+
+                for (final String queryItem : queryItems) {
+                    if (StringUtil.startsWithIgnoreCase(queryItem, PROTOCOL_HANDLER_URL_PARAM)) {
+                        final String value = queryItem.substring(PROTOCOL_HANDLER_URL_PARAM.length());
+                        System.setProperty(PROTOCOL_HANDLER_URL_PROPERTY, value);
+                        repoUrlAvailable = tryParseGitRepoUrl(value);
+                    } else if (StringUtil.startsWithIgnoreCase(queryItem, PROTOCOL_HANDLER_ENCODING_PARAM)) {
+                        final String value = queryItem.substring(PROTOCOL_HANDLER_ENCODING_PARAM.length());
+                        System.setProperty(PROTOCOL_HANDLER_ENCODING_PROPERTY, value);
+                    } else if (StringUtil.startsWithIgnoreCase(queryItem, PROTOCOL_HANDLER_BRANCH_PARAM)) {
+                        final String value = queryItem.substring(PROTOCOL_HANDLER_BRANCH_PARAM.length());
+                        System.setProperty(PROTOCOL_HANDLER_BRANCH_PROPERTY, value);
+                    }
+                }
+
+                if (repoUrlAvailable) {
+                    return true;
+                } else {
+                    found = false;
+                }
+            } else if (arg.equalsIgnoreCase(PROTOCOL_HANDLER_ARG)) {
+                found = true;
+            }
+        }
+
+        return false;
+    }
+
+    public static boolean hasProtocolHandlerRequest() {
+        final String url = System.getProperty(PROTOCOL_HANDLER_URL_PROPERTY, null);
+        return !StringUtil.isNullOrEmpty(url);
+    }
+
+    public static String getProtocolHandlerServer() {
+        return System.getProperty(PROTOCOL_HANDLER_SERVER_PROPERTY, StringUtil.EMPTY);
```

Comment:
It seems strange to me to store these values in the global System properties map since this is a Singleton class and you use the getters to get the values anyway.